### ISSUE TITLE
[Rule Request]: Duplicate Enum Cases

### ISSUE THREAD

**Issue by xavierLowmiller** [realm/SwiftLint#2676]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### New rule request

In Swift, it's currently legal to define two identical enum cases as long as the label of an associated value is different:

```swift
// This compiles
enum PictureImport {
    case add(image: UIImage)
    case add(data: Data)
}
```

This, however, leads to compiler errors when switching:

```swift
// This doesn't compile
func importImage(_ importedImage: PictureImport) {
    switch importedImage {
    case .add(image: let image): // Error: Tuple pattern element label 'image' must be 'data'
        // ...
    case .add(data: let data):
        // ...
    }
}
```

I'd like to propose a rule that warns against identical enum case names.

#### Triggers:

```swift
enum PictureImport {
    case add(image: UIImage)
    case add(data: Data)
}
```

#### Does Not Trigger:

```swift
enum PictureImport {
    case addImage(UIImage)
    case addData(Data)
}
```

In my opinion, this rule should be enabled by default and be marked as an error.

**Comment by jpsim** [comment]
I’m in favor of adding this as a rule.

However, this looks like it might be a Swift bug. Have you reported this at bugs.swift.org?

**Comment by xavierLowmiller** [comment]
Not yet, but I just did: https://bugs.swift.org/browse/SR-10077

Let's see what they have to say.

**Comment by jpsim** [comment]
Thanks for filing that! Curious to see what Swift folks say.
